packages/connect-fastify/src/fastify-connect-plugin.ts

```diff
@@ -60,6 +74,16 @@ export function fastifyConnectPlugin(
   if (opts.acceptCompression === undefined) {
     opts.acceptCompression = [compressionGzip, compressionBrotli];
   }
+  if (opts.shutdownTimeoutMs !== undefined) {
+    const shutdownController = createLinkedAbortController(opts.shutdownSignal);
+    opts.shutdownSignal = shutdownController.signal;
+    instance.addHook("preClose", (done) => {
+      setTimeout(() => {
+        shutdownController.abort(opts.shutdownError);
+      }, opts.shutdownTimeoutMs);
+      done();
```

Comment:
Shouldn't `done` be called _after_ the grace period? So in the setTimeout callback?

Reply:
Either should be fine, Fastify waits for the requests to finish even there are no `preClose`hooks.